inc/listeners/heartbeat.h

```diff
@@ -0,0 +1,91 @@
+#ifndef HEARTBEAT_H
+#define HEARTBEAT_H
+
+#include <iostream>
+#include <deque>
+#include <vector>
+#include <chrono>
+#include <fmt/chrono.h>
+
+#include "events.h"
+#include "instruction.h"
+
+class Heartbeat {

```

Comment:
If you add a constructor here that takes a `std::ostream&`, then we can control the print location.